vipermonkey/core/identifiers.py

```diff
@@ -136,10 +136,13 @@
 # TODO: temporary hack to support any single identifier or object.attrib
 # TODO: support several dots? (object.attrib1.attrib2 etc)
 
-# TODO_identifier_or_object_attrib = Combine(NotAny(reserved_keywords)
-#     + (Combine(Optional(unrestricted_name) + Literal('.') + unrestricted_name) | entity_name))
-
 # TODO: simplified version, using only entity_name
-TODO_identifier_or_object_attrib = Combine(NotAny(reserved_keywords)
-    + Optional(Optional(entity_name) + Literal('.')) + Optional(entity_name + Literal('.')) + entity_name)
+TODO_identifier_or_object_attrib = Combine(NotAny(reserved_keywords) +
+                                           Optional(Optional(entity_name) + Literal('.')) +
+                                           Optional(entity_name + Literal('.')) +
+                                           entity_name +
+                                           # Looks like variables can end in $
+                                           Optional(CaselessLiteral('$')) +
+                                           # Looks like variables can end in '()'.
+                                           Suppress(Optional(CaselessLiteral('()'))))
```

Comment:
Not sure about that one... does not seem to be the best place to remove parentheses. What if it's a function call?

Reply:
I was having trouble parsing certain maldocs a few months ago that from what I remember had assignment statements like 'foo() = 12'. I have removed the '()' from the above parse expression in the current version as the recent maldocs I have been working and a simple test macro all with parse/emulate the same regardless of whether the '()' is included or not in this parse expression.

This change is in commit c963b624223cefed44fbb08bf7e70bbca0a6bf54.

Reply:
OK thanks.